dbms/src/Storages/Distributed/DistributedBlockOutputStream.h

```diff
@@ -21,10 +30,12 @@ class StorageDistributed;
 class DistributedBlockOutputStream : public IBlockOutputStream
 {
 public:
-    DistributedBlockOutputStream(StorageDistributed & storage, const ASTPtr & query_ast, const ClusterPtr & cluster_);
+    DistributedBlockOutputStream(StorageDistributed & storage, const ASTPtr & query_ast, const ClusterPtr & cluster_, bool insert_sync_, UInt64 insert_timeout_ = 0);
 
     void write(const Block & block) override;
 
+    void writePrefix() override { deadline = std::chrono::system_clock::now() + std::chrono::seconds(insert_timeout); }
```

Comment:
Deadline is initialized twice. Do we have any places where `writePrefix` is not called?
Maybe we could remove useless and misleading initialization from constructor.